### PR TITLE
Minor refactoring and replaced deprecated calls and replaced `javax.annotations`

### PR DESCRIPTION
**Pull request by StefanSpieker**
Minor refactoring
* replaced deprecated calls and 
* replaced `javax.annotations` with Spotbugs annotations
* Diamond operator
* minor improvements

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

**Review by pyieh — APPROVED** [review]
LGTM

**Review by MarkEWaite — APPROVED** [review]
Change is ready to merge.

**Comment by MarkEWaite** [comment]
@pyieh since you are a maintainer of this plugin, I believe that you are the right person to merge several pull requests.  

I was prompted by [JENKINS-70460](https://issues.jenkins.io/browse/JENKINS-70460) (ultimately unrelated, but got me started) and submitted several pull requests that would improve the plugin and bring it up to Java 11.  As far as I can tell, the following pull requests are ready to merge:

* #29 
* #30 
* #31 
* #32 
* #33 
* #34

Once those are merged, I believe a little more effort could make the continuous delivery pull request ready to merge as well

* #28 

If you'd like my help with that pull request, I am happy to advise.

**Pull request merged**
